float32 q
float32 q_dot
float32 curr_i